pkg/cmd/experimental/syncgroups/groupsyncer.go

```diff
@@ -141,39 +134,31 @@ func (s *LDAPGroupSyncer) findOpenShiftGroup(ldapGroupUID string) (*userapi.Grou
 
 	group, err := s.GroupClient.Get(groupName)
 	if kapierrors.IsNotFound(err) {
-		if s.SyncExisting {
-			return nil, fmt.Errorf("could not get group for name: %s", groupName)
-
-		} else {
-			newGroup := &userapi.Group{
-				ObjectMeta: kapi.ObjectMeta{
-					Name: groupName,
-					Annotations: map[string]string{
-						ldaputil.LDAPURLAnnotation: s.Host,
-						ldaputil.LDAPUIDAnnotation: ldapGroupUID,
-					},
-				},
-			}
-
-			group, err = s.GroupClient.Create(newGroup)
-			if err != nil {
-				return nil, fmt.Errorf("could not create new group for name %s: %v", groupName, err)
-			}
+		group = &userapi.Group{}
+		group.Name = groupName
+		group.Annotations = map[string]string{
+			ldaputil.LDAPURLAnnotation: s.Host,
+			ldaputil.LDAPUIDAnnotation: ldapGroupUID,
 		}
+
 	} else if err != nil {
 		return nil, err
 	}
 
-	url, exists := group.Annotations[ldaputil.LDAPURLAnnotation]
-	if !exists || url != s.Host {
-		return nil, fmt.Errorf("group %s's %s annotation did not match sync host: wanted %s, got %s",
+	// make sure we aren't taking over an OpenShift group that is already related to a different LDAP group
+	if url, exists := group.Annotations[ldaputil.LDAPURLAnnotation]; !exists || (url != s.Host) {
+		return nil, fmt.Errorf("group %q: %s annotation did not match sync host: wanted %s, got %s",
 			group.Name, ldaputil.LDAPURLAnnotation, s.Host, url)
 	}
-	uid, exists := group.Annotations[ldaputil.LDAPUIDAnnotation]
-	if !exists || uid != ldapGroupUID {
-		return nil, fmt.Errorf("group %s's %s annotation did not match LDAP UID: wanted %s, got %s",
+	if uid, exists := group.Annotations[ldaputil.LDAPUIDAnnotation]; !exists || (uid != ldapGroupUID) {
+		return nil, fmt.Errorf("group %q: %s annotation did not match LDAP UID: wanted %s, got %s",
```

Comment:
> Do we need to change anything else here to support that migrating LDAP group case you are worried about?

Maybe, but I'll keep that change separate.  I think it might be more easily done with a scripted `patch`.